Enterprise-CMCS/macpro-mdct-qmr
# CMDCT-3790: Fixing a few different bugs in involving validation for other measurement specification rate calcs

### Description
There was an issue where for certain Health Home measures (AIF-HH and IU-HH to be specific) if you selected "Other" for Measurement Specification, filled out the form and filled out some of the Optional Measure Stratification (OMS) rates, the measure would save successfully, but it would fail to complete with an error in the console:
<img width="297" alt="Screenshot 2024-07-12 at 1 29 37 PM" src="https://github.com/user-attachments/assets/d67b91c2-f24f-4abc-8d5a-17169d40b3d1">

The cause of this bug is that during the validation process, we were attempting to iterate on an object and the object we were trying to iterate over never even existed in the first place. 

I just removed the offending object iteration because it wasn't really being used anyway. I then changed the way that we grab the optional measure stratification calculations to be validated. This fixed the console error and now the measure can be completed successfully and validates properly. 

---
### How to test

1. Sign into a state that has Health Home coreset ([stateuserNC@test.com](mailto:stateuserNC@test.com) or [stateuserDC@test.com](mailto:stateuserDC@test.com) are two that I know of)
5. Click on the Health Home coreset and navigate to either the AIF-HH or the IU-HH measure
6. Fill out the form with "Other" selected for Measurement Specification
7. When you get to the Optional Measure Stratification, fill out at least one of the fields for Race, Ethnicity, etc.
8. Validate the measure and make sure there is no console error like the one in the description
9. Complete the measure and make sure that the measure truly completes without the console error in the description
10. do some additional testing to make sure that validation for the optional measure stratification rates still works as expected
11. repeat steps 1-8 for 2023 (i'm pretty sure only DC user has HH for '23)
